suites/apis/centralizedAuth.js

```diff
@@ -0,0 +1,655 @@
+   the permissions they have, including RBAC information.
+
+4) As the owner of Kids First, I want to be able to specify AND logic in record ACLs,
+   e.g. C1 AND C2 AND C3 where Cn is a project, in order to guarantee that the person
+   accessing some family-based files from multiple biospecimens has all of the consents.
+
+---------
+Test Plan
+---------
+
+1) - Test that user without policies can't CRUD records in /gen3 or /abc
+
+   - Test user with `abc-admin` policy, test creating record, reading record, updating
+     record, and deleting in indexD with rbac /abc (should work),
+     make sure they still can't create, read, update OR delete in /gen3
+
+   - Test the same thing as above, but use a client's access_token for that user
+
+2) - Test CLIENTB creating signed urls for data under /abc, ensure they cannot create
+     signed urls for data under /gen3. Ensure that successful signed urls actually
+     allow reading data.
+
+   - Test client WITH access creating signed urls for data for user WITHOUT access in namespace.
+
+   - Test client WITHOUT access creating signed urls for data for user WITH access in namespace.
+
+   - Test that a user with `abc-admin` can create signed urls with their own access_token
+
+3) - Test client flow to get id_token, compare to what's in userinfo endpoint, ensure
+     RBAC policy information is there
+        > NOTE: there is an existing test that might satisfy this with a few updates
+
+4) - Create some records in indexd with AND logic, have CLIENTB try to create signed
+     url for file for user that does NOT have necessary permissions, ensure failure.
+
+   - Same as above but user DOES have permissions. Ensure successfull signed URL that
+     we can access data with
+
+
+We probably want some tests to make sure that after usersyncing and changing a user's
+policies, they can no longer create signed urls?
+
+dbgap syncing tests would be good too (since all these rely on user.yaml)
+
+------------------------------------
+Default Authorization Details
+------------------------------------
+
+Access details from defualt usersync:
+
+Create CLIENT with `abc-admin` and `gen3-admin` policy (CRUD under /abc and /gen3)
+Create CLIENTB with `abc-admin` policy (CRUD under /abc)
+
+cdis.autotest@gmail.com (mainAcct)
+    - abc-admin
+
+dummy-one@planx-pla.net (auxAcct1)
+    - abc.programs.test_program.projects.test_project1-viewer
+
+smarty-two@planx-pla.net (auxAcct2)
+    - abc.programs.test_program2.projects.test_project3-viewer
+
+dcf-integration-test-0@planx-pla.net (user0)
+    - gen3-admin
+*/
+const { Commons } = require('../../utils/commons.js');
+const chai = require('chai');
+const { Bash } = require('../../utils/bash.js');
+const apiUtil = require('../../utils/apiUtil.js');
+const uuid = require('uuid');
+
+const bash = new Bash();
+
+const fs = require('fs');
+
+const I = actor();
+
+/*
+for twoProjectsFile, use cdis.autotest@gmail.com
+
+Each has ONE of these policies:
```

Comment:
i find this comment a bit confusing, does it mean "for twoProjectsFile: cdis.autotest@gmail.com has access to both projects, dummy-one and smarty-two have access to one each"?

Reply:
yeah, this is sorta outdated. I will remove since I added details about access in the comments above